common/dynamicconfig/constants.go

```diff
@@ -2120,456 +1224,3736 @@ const (
+		},
+		DefaultValue: time.Minute * 10,
+	},
+	ESAnalyzerTimeWindow: DynamicDuration{
+		DynamicBase: DynamicBase{
+			KeyName:     "worker.ESAnalyzerTimeWindow",
+			Description: "ESAnalyzerTimeWindow defines the time window ElasticSearch Analyzer will consider while taking workflow averages",
+		},
+		DefaultValue: time.Hour * 24 * 30,
+	},
+	ESAnalyzerBufferWaitTime: DynamicDuration{
+		DynamicBase: DynamicBase{
+			KeyName:     "worker.ESAnalyzerBufferWaitTime",
+			Description: "ESAnalyzerBufferWaitTime controls min time required to consider a worklow stuck",
+		},
+		DefaultValue: time.Minute * 30,
+	},
+}
+
+var MapKeys = map[MapKey]DynamicMap{
+	TestGetMapPropertyKey: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "testGetMapPropertyKey",
+			Description: "",
+		},
+		DefaultValue: nil,
+	},
+	RequiredDomainDataKeys: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "system.requiredDomainDataKeys",
+			Description: "RequiredDomainDataKeys is the key for the list of data keys required in domain registration",
+		},
+		DefaultValue: nil,
+	},
+	ValidSearchAttributes: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "frontend.validSearchAttributes",
+			Description: "ValidSearchAttributes is legal indexed keys that can be used in list APIs. When overriding, ensure to include the existing default attributes of the current release",
+		},
+		DefaultValue: definition.GetDefaultIndexedKeys(),
+	},
+	TaskSchedulerRoundRobinWeights: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "history.taskSchedulerRoundRobinWeight",
+			Description: "TaskSchedulerRoundRobinWeights is the priority weight for weighted round robin task scheduler",
+		},
+		DefaultValue: common.ConvertIntMapToDynamicConfigMapProperty(map[int]int{
+			common.GetTaskPriority(common.HighPriorityClass, common.DefaultPrioritySubclass):    500,
+			common.GetTaskPriority(common.DefaultPriorityClass, common.DefaultPrioritySubclass): 20,
+			common.GetTaskPriority(common.LowPriorityClass, common.DefaultPrioritySubclass):     5,
+		}),
+	},
+	QueueProcessorPendingTaskSplitThreshold: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "history.queueProcessorPendingTaskSplitThreshold",
+			Description: "QueueProcessorPendingTaskSplitThreshold is the threshold for the number of pending tasks per domain",
+		},
+		DefaultValue: common.ConvertIntMapToDynamicConfigMapProperty(map[int]int{0: 1000, 1: 10000}),
+	},
+	QueueProcessorStuckTaskSplitThreshold: DynamicMap{
+		DynamicBase: DynamicBase{
+			KeyName:     "history.queueProcessorStuckTaskSplitThreshold",
+			Description: "QueueProcessorStuckTaskSplitThreshold is the threshold for the number of attempts of a task",
+		},
+		DefaultValue: common.ConvertIntMapToDynamicConfigMapProperty(map[int]int{0: 100, 1: 10000}),
+	},
+}
+
+var _keyNames map[string]Key
+
+func init() {
+	_keyNames = make(map[string]Key)
+	for k, v := range IntKeys {
+		_keyNames[v.KeyName] = k
```

Comment:
should verify that they're all unique here, since it's kinda our only chance to do so

Reply:
yes, we should do it, will make an update